lib/linter/node-event-generator.js

```diff
@@ -293,6 +308,12 @@ class NodeEventGenerator {
      * @returns {void}
      */
     applySelector(node, selector) {
+
+        if (this.knownTypes.has(selector.rawSelector)) {
+            this.emitter.emit(selector.rawSelector, node);
+            return;
+        }
+
         if (esquery.matches(node, selector.parsedSelector, this.currentAncestry, this.esqueryOptions)) {
```

Comment:
Can we merge this condition with the next one in order to avoid repeating `this.emitter.emit(selector.rawSelector, node);`?

Reply:
Hmm, I think you meant to post this on line 312?

I prefer the way I have it written because it's easier to see that there are two distinct conditions being handled. I don't mind repeating that line.

Reply:
> Hmm, I think you meant to post this on line 312?

Yes, I should have clicked on that line rather than the whole changed block, sorry.

> I prefer the way I have it written because it's easier to see that there are two distinct conditions being handled. I don't mind repeating that line.

Fair enough. That would also not affect performance, only style.